### PR TITLE
[IMP] l10n_si: update Slovenian localization

### PR DESCRIPTION
**Pull request by rdeodoo**
- Update CoA
- Update taxes
- Update tax groups
- Update reports
- Update fiscal positions

Courtesy of `Editor.si` for providing data in order to update the files.

Enterprise PR: odoo/enterprise#65817

Task [link](https://www.odoo.com/odoo/project/967/tasks/3901247)
task-3901247

### CODE REVIEW

File: addons/l10n_si/data/template/account.tax-si.csv
Comment (lordkrandel) [inline addons/l10n_si/data/template/account.tax-si.csv:66]: "0% Exit VAT 3. world"? Does this mean "0% Export outside EU?"
I think we should review these English translations to follow our own norms.

File: addons/l10n_si/data/template/account.tax-si.csv
Comment (lordkrandel) [inline addons/l10n_si/data/template/account.tax-si.csv:62]: Does `output` mean `export` here?

File: addons/l10n_si/data/template/account.tax.group-si.csv
Comment (lordkrandel) [inline addons/l10n_si/data/template/account.tax.group-si.csv:9]: Please put the tax ids in order, (move the `tax_group_8` over tax_group_5`) since we're changing everything I don't think the diff size matters.

Rename one of the two:
- `tax_group_so_05` -> `tax_group_so_5`
- `tax_group_5` -> `tax_group_05`

What are these SO groups btw? Samo ocenjevanje? Self-assessment, self taxation... sounds like it's Reverse Charge. In that case please follow the [Tax Naming Conventions](https://www.odoo.com/odoo/knowledge.article/1518?cids=1), it should be "RC" then.

image

**Review by lordkrandel — CHANGES_REQUESTED** [review]
It seems as the PO is still to test this, so I'm writing some observation that leap up to my eye at the moment. Let me know what you all think.

- Please make a more fluent PR description *and* commit message.
It's important to stay readable, it's useful for future use and it also helps improving communication skills.

```
In collaboration with our partner ﻿[﻿Editor.si﻿](http://editor.si/)﻿, we have revised the basic localization package.

- Accounts in the CoA
- Tax definitions
- Tax groups
- Tax report
- Fiscal Positions

[Task link](https://www.odoo.com/odoo/project.task/3901247)
task-3901247
```

In the commit message you can skip the links for the task and the partner.

- I'm also leaving observations on how the names, descriptions, data given by the partner is integrated in Odoo. These things are better to be detailed now than later ^^

**Comment by lordkrandel** [comment]
Guaps. I've just realized that probably I should @malb-odoo guide you through this 😄

**Comment by rdeodoo** [comment]
Thanks for the review.

Note that as stated on the task:
- I won't be working on this PR anymore. Another dev will finish the work. I'll let you check on the chatter for in-depth detail about it.
- There is mainly translations to finish and tax short codes which I did not have time to do before leaving the team

For the translations (as you had some questions about it), note that the partner did not provide any English terms, only Slovenian wording. So the English terms in the PR are auto-translated terms as requested by the PO (I improved some after auto-translation but still, you got the idea).

> In the commit message you can skip the links for the task and the partner.

Note that having the task link in the commit message is generally far more important than in the PR, so it should be in both but especially in the commit message, for blaming purpose (where the PR has low value): it's better to be able to go to the task directly without having to open the PR.

**Comment by lordkrandel** [comment]
> Note that having the task link in the commit message is generally far more important than in the PR, so it should be in both but especially in the commit message, for blaming purpose (where the PR has low value): it's better to be able to go to the task directly without having to open the PR.

I don't agree 😄 
- The commit has always a direct link to the PR that merged it. (`closes #xxxxxx` )
- Clicking on the PR is no cost at all.
- The task is related to a PR, not to a commit, especially in case of multi-commit changes.
- The PR has a direct link to its commit, but the commit cannot have a direct link to the task, just the number.

So PRs win IMHO, and furthermore, I think it's important to have the task name in both. 😛

**Comment by rdeodoo** [comment]
> The task is related to a PR (or even a bundle), not to a commit, especially in case of multi-commit changes.

Indeed, and the commit is related to the task :)

> Clicking on the PR is no cost at all.

Depends of your flow, when you are blaming files in `tig` or whatever, I don't want to go through github, I just want to copy paste the task id to open it directly.

It's exactly the same reason as why people are complaining about commit messages being too long (should never overflow 72/80 char, but going over 100 should really be a no go), because you absolutely have no way to read it in `tig` and so on when blaming. You have to scroll right, then left, then bottom, then repeat that for every sentence. A bit like old typewriters :grimacing: 

Luckily, most of the R&D now enforced that guideline, and if I am not wrong we finally managed to have the runbot make it mandatory since a few weeks?

Just my 2 cents, but IMHO PR is a just a bonus, commit msg should contains everything, and if the PR does too, it's perfect.

**Comment by lordkrandel** [comment]
I'm not aware of the workflow with `tig`, will have a look, thanks!

> Just my 2 cents, but IMHO PR is a just a bonus, commit msg should contains everything, and if the PR does too, it's perfect.

Agreed 🤝

**Comment by malb-odoo** [comment]
@lordkrandel I will not do the review on this one, i will finish the pr so it will follow the flow like a normal PR

File: addons/l10n_si/data/template/account.tax-si.csv
Comment (malb-odoo) [reply addons/l10n_si/data/template/account.tax-si.csv:62]: Output means sale i think

File: addons/l10n_si/data/template/account.tax-si.csv
Comment (malb-odoo) [reply addons/l10n_si/data/template/account.tax-si.csv:66]: i talk to christophe about that and he said to put EX (export) instead of world

File: addons/l10n_si/data/account_tax_report_data.xml
Comment (malb-odoo) [reply addons/l10n_si/data/account_tax_report_data.xml:424]: I think the tax report are put as document in the task :smile:, anyway the partner will soon test the branch :smile:

File: addons/l10n_si/data/template/account.tax-si.csv
Comment (lordkrandel) [reply addons/l10n_si/data/template/account.tax-si.csv:66]: `0% EX G`, yet I don't get what the `3.` stands for.

Comment (malb-odoo) [reply addons/l10n_si/data/template/account.tax-si.csv:66]: Maybe for third world countries ? but not sure about that, i could just remove it but since it's a description i don't think we really care

File: addons/l10n_si/data/template/account.tax-si.csv
Comment (malb-odoo) [reply addons/l10n_si/data/template/account.tax-si.csv]: That's what i put in the name yes :smile: i think the description can stay like that

Comment (lordkrandel) [reply addons/l10n_si/data/template/account.tax-si.csv]: Ok for the name, but I would really remove "output" from the descrpition, or change it to "sale". It's unnecessary and confusing.

File: addons/l10n_si/data/template/account.tax-si.csv
Comment (lordkrandel) [reply addons/l10n_si/data/template/account.tax-si.csv:66]: Also here, I'd say "Export" instead of "Exit" and remove the "3.". since the name "0% EX" doesn't hint at anything about peculiar "third world" incentive taxes.

We as Odoo developers are the primary users of those descriptions - Slovenian businesses will use the translated ones, so I do care, unclear descriptions make it up for confusion later.

**Comment by malb-odoo** [comment]
@lordkrandel Could you re-review this one and the enterprise pr when you have time :smile:

**Comment by lordkrandel** [comment]
Oook.

**Review by lordkrandel — CHANGES_REQUESTED** [review]
A first look on taxes 😄

**Review by lordkrandel — CHANGES_REQUESTED** [review]
For every description we try to conform to our standards in English (like putting the VAT rate always in front of the tax description and not 80% of the time) we will have to do the same in Slovenian language.

File: addons/l10n_si/data/template/account.tax-si.csv
Comment (malb-odoo) [reply addons/l10n_si/data/template/account.tax-si.csv:105]: Asked Christophe about it

File: addons/l10n_si/data/account_tax_report_data.xml
Comment (malb-odoo) [reply addons/l10n_si/data/account_tax_report_data.xml:962]: That's the translation of what the partner gave us, i suppose it's referring to a section or chapter in a legal document, As of now we don't really have more information, i think we can ship it and correct that later if needed :smile:

File: addons/l10n_si/data/account_tax_report_data.xml
Comment (malb-odoo) [reply addons/l10n_si/data/account_tax_report_data.xml:826]: That's the translation of what the partner gave us, i can check with christophe if we want to change the name, but i think it's better to ship the task and see later

**Comment by lordkrandel** [comment]
If you manage to solve these points we found already, then I rely on CHKL to validate it functionally. Since it's mostly data, I don't know if it adheres to what a Slovenian accountant expects, the code looks ok.

File: addons/l10n_si/data/account_tax_report_data.xml
Comment (lordkrandel) [reply addons/l10n_si/data/account_tax_report_data.xml:826]: Make a follow-up task then, so we dont lose track.

**Comment by malb-odoo** [comment]
@lordkrandel just talked to christophe about the tax and the reports.
For the tax: It's normal, if you look at the repartitions lines it's a 0% tax but we need to indicate that this is a 22% one on the invoice.
For the report: It's more of a generic problem of the report. The report should take more space so that we don't have the ellipsis.

He told me to ship it like this and we can come back later on :smile:

**Review by lordkrandel — APPROVED** [review]
LGTM 👍🏻 

@smetl for final review (or if you can find someone else available!)

File: addons/l10n_si/data/account_tax_report_data.xml
Comment (rugo-odoo) [inline addons/l10n_si/data/account_tax_report_data.xml:484]: Use english record ids. Easier to understand :smile: (Happens several times on file)

File: addons/l10n_si/data/account_tax_report_data.xml
Comment (rugo-odoo) [inline addons/l10n_si/data/account_tax_report_data.xml:592]: Break in multiple lines.

File: addons/l10n_si/data/account_tax_report_data.xml
Comment (rugo-odoo) [inline addons/l10n_si/data/account_tax_report_data.xml:3]: Split each report into a file.

**Review by rugo-odoo — CHANGES_REQUESTED** [review]
Other than that, it's good to go.

**Comment by rugo-odoo** [comment]
@robodoo r+

**Comment by rdeodoo** [comment]
FYI, I did not follow what happened here since I moved out of RD, but since @malb-odoo took over and finished it, it would have been more than fine to add himself as co-author in case he wanted?
Looks like it was not done, but maybe worth considering should the staging fail and need another r+ :)

**Comment by rugo-odoo** [comment]
> FYI, I did not follow what happened here since I moved out of RD, but since @malb-odoo took over and finished it, it would have been more than fine to add himself as co-author in case he wanted? Looks like it was not done, but maybe worth considering should the staging fail and need another r+ :)

He doesn't care about it, it was suggested.

**Pull request closed without merging**
